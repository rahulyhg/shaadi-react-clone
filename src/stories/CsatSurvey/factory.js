const data = {
  match: {
    path: '/stop-page/:pageName',
    url: '/stop-page/csat-survey',
    isExact: true,
    params: {
      pageName: 'csat-survey',
    },
  },
  layout: 'desktop',
  history: {
    length: 7,
    action: 'POP',
    location: {
      pathname: '/stop-page/csat-survey',
      search: '',
      hash: '',
    },
  },
  location: {
    pathname: '/stop-page/csat-survey',
    search: '',
    hash: '',
  },
  isLoggedOut: false,
  isNative: false,
  pageName: 'csat-survey',
  user: {
    isLoggedIn: true,
    isLoggedOut: false,
    errusr: '',
    uid: 'vSH88709866',
    gender: '',
    firstName: '',
    lastName: '',
    religion: '',
    motherTongue: '',
    country: '',
    complexion: '',
    built: '',
    height: '',
    weight: '',
    diet: '',
    drink: '',
    smoke: '',
    aboutMe: '',
    hasProfilePhoto: false,
    photos: {},
    heShe: '',
    himHer: '',
    hisHer: '',
    mrMs: '',
    heSheOrYou: '',
    himHerOrYou: '',
    hisHerOrYou: '',
    mrMsOrYou: '',
    heSheOrYour: '',
    himHerOrYour: '',
    hisHerOrYour: '',
    mrMsOrYour: '',
    doOrDoes: '',
    ptnr: 'direct_homepage',
    sessionId: '6BC042D1E12D335D756D56A469B435FB1522823161vSH88709866',
  },
  nextUrl: '/my-shaadi',
  csatSurvey: {
    questions: [
      {
        id: '1',
        display_order: '1',
        type: 'multichoice',
        title: 'What do you like about Shaadi.com?',
        placeHolder: '',
        isRequired: false,
        choices: {
          '20': 'Easy to Use',
          '21': 'Quality of Matches',
          '22': 'Lots of Matches',
        },
      },
      {
        id: '2',
        display_order: '3',
        type: 'textarea',
        title: 'Can you tell us what can we do better?',
        placeHolder: 'Write your feedback here',
        isRequired: false,
        choices: {},
      },
      {
        id: '3',
        display_order: '2',
        type: 'rating',
        title: 'Please rate your Experience with us',
        placeHolder: '',
        isRequired: false,
        choices: {
          '1': 'Hated it',
          '2': 'Disliked it',
          '3': "It's OK",
          '4': 'Liked it',
          '5': 'Loved it',
        },
      },
    ],
    loading: false,
  },
  wwwBaseUrl: 'https://www.shaadi.com',
  documentUpload: {
    uid: null,
    source: null,
    attachments: {},
    redirectToAlbums: false,
    isModalClosed: false,
    isUploadFailed: false,
    uploadingPhotoCount: 0,
  },
  doCsatSurvey() {},
};

export default data;
