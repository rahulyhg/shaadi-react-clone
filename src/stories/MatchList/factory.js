const props = {
  profileUrl:
    '/profile?profileid=qSH73251586&pg_show_from=1&np=search-result&evt_ref=bWF0Y2hlcy1uZXdfbWF0Y2hlc191bnZpZXdlZA==&navigation=1&profileNumber=2&pg_searchresults_id=search:42efdd941783200f7ab8e98cae39a461&datasrc=api&pg_ubt=L3NlYXJjaC9uZXctbWF0Y2hlc3x8MjA=&featured=N',
  albumUrl:
    'https://www.shaadi.com/profile/index/view-album-photos/profileid/qSH73251586/navigation/1/profileNumber/2/evt_ref/bWF0Y2hlcy1uZXdfbWF0Y2hlc191bnZpZXdlZA==/pg_searchresults_id/search:42efdd941783200f7ab8e98cae39a461',
  isSpotLight: false,
  isSelected: false,
  isHovered: false,
  isNewMatchHovered: false,
  settings: {
    defaultSearchFormat: 'list',
    isPaidUser: false,
    showUpgradeBanner: true,
    canAccessChat: true,
    canInitiateChat: false,
    hasUploadedPhoto: false,
    canConnectWithMessage: false,
    showUpgradeLinks: true,
    canSendPasswordOnConnect: false,
    contactsTotal: null,
    contactsRemaining: null,
    isUnderScreening: false,
    isAstroGamified: true,
    gender: 'Female',
    canViewCollegeAndEmployer: false,
    canViewHoroscope: true,
    horoscopeStyle: 'l/ENG/hs/1',
    isFamilyGamified: true,
    isHidden: false,
    isBothPartyPayUser: false,
    isMobileVerified: false,
    mobileNumber: '9379732670',
    wasPaidUser: false,
    expiryDate: '',
    isNri: false,
    isIndianDiaspora: true,
    experiments: {
      premium_tag: {
        bucket: 'A',
      },
      near_me: {
        bucket: 'A',
      },
      webCTA_Dec2017: {
        bucket: 'B',
      },
      premium_carousel: {
        bucket: 'A',
      },
      cartpage: {
        bucket: 'A',
      },
      profile_verification: {
        bucket: 'B',
      },
      offline_otp: {
        bucket: 'B',
      },
      inbox_revamp: {
        bucket: 'A',
      },
      track_events: {
        bucket: 'A',
      },
    },
    membershipTags: 'free',
    offerCode: '',
    offer_details: '',
  },
  profile: {
    base: {
      infoMap: [
        {
          key: 'info-0',
          label: 'Age / Height',
          value: '28, 5\' 8"',
        },
        {
          key: 'info-1',
          label: 'Religion/Community',
          value: 'Hindu, Kashyap',
        },
        {
          key: 'info-2',
          label: 'Mother Tounge',
          value: 'Hindi',
        },
        {
          key: 'info-3',
          label: 'Profession',
          value: 'Software Professional (Others)',
        },
        {
          key: 'info-4',
          label: 'Location',
          value: 'Sydney, Australia',
        },
      ],
      infoList: [
        {
          key: 'age-height',
          value: '28 yrs, 5\' 8", Hindu, Hindi',
        },
        {
          key: 'profession',
          value: 'Software Professional (Others)',
        },
        {
          key: 'location',
          value: 'Lives in Sydney, Australia',
        },
        {
          key: 'grew up in',
          value: 'Grew up in India',
        },
      ],
      detailList: [
        {
          key: 'age-height',
          value: '28, 5\' 8"',
        },
        {
          key: 'religion-caste',
          value: 'Hindu, Kashyap',
        },
        {
          key: 'profession',
          value: 'Software Professional (Others)',
        },
        {
          key: 'location',
          value: 'Sydney, Australia',
        },
      ],
      premiumInfo: [
        {
          key: 'age-height',
          value: '28 yrs, 5\' 8", Hindu, Hindi',
        },
        {
          key: 'location',
          value: 'Sydney, Australia',
        },
      ],
      miniList: [
        {
          key: 'age-height',
          value: '28 yrs, 5\' 8", Hindu, Hindi',
        },
        {
          key: 'religion-caste',
          value: 'Kashyap',
        },
        {
          key: 'location',
          value: 'Lives in Sydney, Australia',
        },
      ],
      miniNriList: [
        {
          key: 'age-height',
          value: '28 yrs, 5\' 8", Hindu, Hindi',
        },
        {
          key: 'grew up in',
          value: 'Grew up in India',
        },
        {
          key: 'location',
          value: 'Lives in Sydney, Australia',
        },
      ],
    },
    flags: {
      isFree: true,
      isPremium: false,
      membershipLevel: 'Free',
      connectionNote: null,
      connectionError: false,
      connectionStatus: 'default',
      connectionAction: 'not_contacted',
      connectionJustNowText: null,
      contactAction: 'free',
      contactStatus: 'availableOnVerification',
      horoscopeStatus: 'none',
      loading: 'none',
      isDeleted: false,
      isNri: true,
      isWatermarked: true,
      isTwoWayMatch: true,
      isBoldListing: false,
      canUnblock: false,
      unblockMessage: null,
      canUnignore: false,
      canRemind: false,
      canCancelInvite: false,
      isSameGender: false,
      activeStatus: 'default',
      isPreferredMatch: false,
      isConnectLimitExceeded: false,
      showHistory: false,
      membershipTags: 'free',
      isNameLocked: true,
      isMaskedProfile: true,
      canCallSendSMS: false,
      canSendEmail: false,
      canSendEmailReminder: false,
      canSendSMS: false,
      canViewPhoneNo: false,
      isPhoneNoViewed: false,
      isSmsAlreadySent: false,
      showChatNow: false,
      showPostOnWall: false,
      isFiltered: false,
      isSkipped: false,
      isProfileViewed: false,
      canViewPhotosOnConnectOnly: false,
      canPremiumMemberViewPhoto: false,
      horoscopeStyle: 'l/undefined/hs/1',
      albumStatus: 'default',
      isIndianDiaspora: true,
      isHoroscopeApplicable: true,
      isFamilyGamified: false,
      isAstroReady: false,
      isHidden: false,
      hiddenReason: null,
      isBothPartyPayUser: false,
    },
    presence: {
      onlineStatus: 'Idle',
      onlineStatusDetails: '...',
      onlineAt: 1521705542000,
      device: 'web',
      platform: 'web',
      lastOnline: 'Last online, 48 mins ago',
      lastOnlineDetails: 'Online 1h ago',
      ready: true,
      chatIcon: 'web_idle',
    },
    summary: {
      infoMap: [
        {
          key: 'age_height',
          label: 'Age / Height',
          value: '28, 5\' 8"',
        },
        {
          key: 'religion',
          label: 'Religion',
          value: 'Hindu',
        },
        {
          key: 'mother_tongue',
          label: 'Mother Tongue',
          value: 'Hindi',
        },
        {
          key: 'community',
          label: 'Community',
          value: 'Kashyap',
        },
        {
          key: 'location',
          label: 'Location',
          value: 'Sydney, Australia',
        },
        {
          key: 'education',
          label: 'Education',
          value: 'Masters - Computers/ IT',
        },
        {
          key: 'profession',
          label: 'Profession',
          value: 'Software Professional (Others)',
        },
      ],
      infoMapNonIndian: [
        {
          key: 'age_height',
          label: 'Age / Height',
          value: '28, 5\' 8"',
        },
        {
          key: 'religion',
          label: 'Religion',
          value: 'Hindu',
        },
        {
          key: 'mother_tongue',
          label: 'Mother Tongue',
          value: 'Hindi',
        },
        {
          key: 'location_nri_info',
          label: 'Location',
          value: 'Sydney,New South Wales, Australia',
        },
        {
          key: 'info-6',
          label: 'Grew up in',
          value: 'India',
        },
        {
          key: 'education',
          label: 'Education',
          value: 'Masters - Computers/ IT',
        },
        {
          key: 'profession',
          label: 'Profession',
          value: 'Software Professional (Others)',
        },
      ],
      infoMapIndian: [
        {
          key: 'age_height_list',
          label: 'Age / Height',
          value: '28 yrs, 5\' 8"',
        },
        {
          key: 'marital_status',
          label: 'Marital Status',
          value: 'Never Married',
        },
        {
          key: 'religion_caste_list',
          label: 'Religion',
          value: 'Hindu, Kashyap',
        },
        {
          key: 'location',
          label: 'Location',
          value: 'Sydney, Australia',
        },
        {
          key: 'mother_tongue',
          label: 'Mother Tongue',
          value: 'Hindi',
        },
        {
          key: 'profession',
          label: 'Profession',
          value: 'Software Professional (Others)',
        },
      ],
      infoMapNri: [
        {
          key: 'age_height_list',
          label: 'Age / Height',
          value: '28 yrs, 5\' 8"',
        },
        {
          key: 'profession',
          label: 'Profession',
          value: 'Software Professional (Others)',
        },
        {
          key: 'religion_mother_tongue_list',
          label: 'Mother Tongue',
          value: 'Hindu, Hindi',
        },
        {
          key: 'location_nri_info_list',
          label: 'Location',
          value: 'Lives in Sydney,New South Wales, Australia',
        },
        {
          key: 'education',
          label: 'Education',
          value: 'Masters - Computers/ IT',
        },
        {
          key: 'grew_up_in_info',
          label: 'Grew up in',
          value: 'Grew up in India',
        },
      ],
      listAlbum: [
        'https://img1.shaadi.com/2018/03/22/qSH73251586-65e46d-male.webp',
        'https://img1.shaadi.com/2018/03/22/qSH73251586-819f9b-male.webp',
        'https://img1.shaadi.com/2018/03/22/qSH73251586-1b3d1f-male.webp',
        'https://img2.shaadi.com/imgs/profiles/150-photo-coming-soon-m.gif',
      ],
      gridAlbum: [
        'https://img1.shaadi.com/2018/03/22/qSH73251586-edb5d5-male.webp',
        'https://img1.shaadi.com/2018/03/22/qSH73251586-fa24af-male.webp',
        'https://img1.shaadi.com/2018/03/22/qSH73251586-fdc10b-male.webp',
        'https://img2.shaadi.com/imgs/profiles/250-photo-coming-soon-m.gif',
      ],
      shortBio: 'Smart, I am living in Australia from last 3 years, currently living in Sydney, Australia and intend to settle here only.',
      createdBy: 'Parent / Guardian',
      profileCreatedBy: 'Parent',
      shortlistCount: 0,
    },
    detailed: {
      album: ['https://img1.shaadi.com/2018/03/22/qSH73251586-edb5d5-male.webp'],
      infoMap: [],
      about: {
        title: 'About',
        icon: 'about',
        desc: '',
      },
      lifestyle: {
        title: 'Lifestyle & Appearance',
        icon: 'lifestyle',
        items: [],
      },
      interests: {
        title: 'Interests & More',
        icon: 'interests',
        items: [],
      },
      background: {
        title: 'Background',
        icon: 'background',
        items: [],
      },
      education: {
        title: 'Education & Career',
        icon: 'education',
        items: [],
        hasCollegeOrEmployer: false,
      },
      horoscope: {
        title: 'Horoscope Details',
        icon: 'astro',
        isGamified: false,
        items: [],
      },
      family: {
        title: 'Family Details',
        icon: 'family',
        isGamified: false,
        desc: '',
      },
      preferences: {
        title: 'Preferences Details',
        icon: 'preferences',
        isGamified: false,
        items: [],
      },
      trustBadges: [],
      commonInterests: [],
    },
    requests: {
      count: 0,
      items: [],
    },
    shortlists: {
      selected: [],
      ready: true,
      count: 0,
    },
    photos: {
      count: 0,
      items: [],
      status: '',
      hasPhotos: false,
      canAddPhotos: false,
      isDefault: true,
      isRejectedPhotosFetched: false,
    },
    astro: {
      details: {},
      chart: {},
      hasAstro: false,
    },
    privacy: {
      display_name: '',
      horoscope_status: '',
      memberlogin: '',
      phone: '',
      photo: '',
      photo_password: null,
      profile_privacy: '',
      shortlist_setting: '',
      sms_alert: 0,
      visitors_setting: '',
    },
    contactSummary: {
      sms: '',
      contact: {},
      actionDate: '',
      dnd: '',
      hidden: '',
      hiddenReason: '',
      tempKey: '',
      contactType: '',
      photoStatus: '',
      photoCount: 0,
      profileContactStatus: '',
      contactDetailsStatusString: '',
      contactDetailsTitleStatusString: '',
    },
    connectMessages: [],
    thumbnailBlur: 'https://img1.shaadi.com/2018/03/22/qSH73251586-85e5b4-male.webp',
    thumbnail: 'https://img1.shaadi.com/2018/03/22/qSH73251586-85e5b4-male.webp',
    photoBlur: 'https://img1.shaadi.com/2018/03/22/qSH73251586-65e46d-male.webp',
    photoMedium: '/assets/default-photo.png',
    photo: 'https://img1.shaadi.com/2018/03/22/qSH73251586-65e46d-male.webp',
    fullPhotoBlur: 'https://img1.shaadi.com/2018/03/22/qSH73251586-edb5d5-male.webp',
    fullPhoto: 'https://img1.shaadi.com/2018/03/22/qSH73251586-edb5d5-male.webp',
    drAction: {},
    heShe: 'He',
    himHer: 'Him',
    hisHer: 'His',
    gender: 'Male',
    tempId: '',
    verification: {
      count: 0,
      shield_state: '',
      derived_text: '',
      verified_proofs: [],
    },
    se: '69477495232f1556b264e7c2f22b1802',
    uid: 'qSH73251586',
    name: 'Sahil C',
    fullName: 'Sahil Chaudhary',
    firstName: 'Sahil',
    lastName: 'Chaudhary',
    slug: 'qSH73251586',
    userHandle: 'SH73251586',
    location: 'Sydney, Australia',
    mrMs: 'Mr.',
  },
  tooltip: {
    key: 'none',
    page: 'none',
    position: 'none',
    title: null,
    body: [],
    loading: false,
  },
  item: {
    uid: 'qSH73251586',
    justNow: false,
    eoiLoadingStyle: 'none',
    photoLoading: false,
    actionType: '',
    displayStatusMessage: '',
  },
  shortlistItems: [
    {
      key: 'sl-6761137',
      id: '6761137',
      label: "My Maybe's",
    },
  ],
  wwwBaseUrl: 'https://www.shaadi.com',
  membershipTags: 'free',
  onAction() {},
  onMouseEnter() {},
  onMouseLeave() {},
  onShowContactDetails() {},
  onChatNow() {},
  onShowWatermarkInfo() {},
  onEoiTooltipClose() {},
  onPhotoTooltipClose() {},
  onRequestPhoto() {},
  onRequestPassword() {},
  onShortlistOpen() {},
  onDirectlyShortlist() {},
  onDirectlyRemoveShortlist() {},
  onDirectlyIgnore() {},
  onNewMatchMouseEnter() {},
  onNewMatchMouseLeave() {},
  daTracking() {},
  onCallConsultantInvited() {},
  onClick() {},
};

export default props;
