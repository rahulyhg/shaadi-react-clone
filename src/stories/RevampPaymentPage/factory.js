import React from 'react';

const goldProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '50% off',
  duration: (
    <React.Fragment>
      <span>3 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: false,
  name: 'Gold',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 742,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 4450,
  productCode: 'SSP_G3',
  revampPage: false,
  saleprice: 2225,
  tagName: '',
  topAmountProps: {
    amount: 2225,
    currency: 'INR',
  },
};
const goldPlusProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '55% off',
  duration: (
    <React.Fragment>
      <span>3 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: false,
  name: 'Gold Plus',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 833,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 5550,
  productCode: 'SSP_GPlus',
  revampPage: false,
  saleprice: 2497,
  tagName: '',
  topAmountProps: {
    amount: 2497,
    currency: 'INR',
  },
};
const diamondProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '55% off',
  duration: (
    <React.Fragment>
      <span>6 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: false,
  name: 'Diamond',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 491,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 6550,
  productCode: 'SSP_D6',
  revampPage: false,
  saleprice: 2947,
  tagName: '',
  topAmountProps: {
    amount: 2947,
    currency: 'INR',
  },
};
const diamondPlusProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '60% off',
  duration: (
    <React.Fragment>
      <span>6 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: false,
  name: 'Diamond Plus',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 577,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 8650,
  productCode: 'SSP_DPlus',
  revampPage: false,
  saleprice: 3460,
  tagName: 'top_seller',
  topAmountProps: {
    amount: 3460,
    currency: 'INR',
  },
};
const platinumPlusProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '60% off',
  duration: (
    <React.Fragment>
      <span>12 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: false,
  name: 'Platinum Plus',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 488,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 14650,
  productCode: 'SSP_PPlus',
  revampPage: false,
  saleprice: 5860,
  tagName: 'best_value',
  topAmountProps: {
    amount: 5860,
    currency: 'INR',
  },
};
const yourPlanProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '55% off',
  duration: (
    <React.Fragment>
      <span>3 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: true,
  isSelected: true,
  name: 'Gold Plus',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 833,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 5550,
  productCode: 'SSP_GPlus',
  revampPage: false,
  saleprice: 2497,
  tagName: 'your_plan',
  topAmountProps: {
    amount: 2497,
    currency: 'INR',
  },
};
const selectThreeMonthProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '55% off',
  duration: (
    <React.Fragment>
      <span>3 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: false,
  isSelected: false,
  name: 'select 3',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 29500,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 29500,
  productCode: 'PP_SP1',
  revampPage: false,
  saleprice: 29500,
  tagName: '',
  topAmountProps: {
    amount: 9834,
    currency: 'INR',
  },
};
const selectSixMonthProps = {
  btnloading: false,
  currency: 'INR',
  discountDetails: '',
  discountText: '55% off',
  duration: (
    <React.Fragment>
      <span>6 months</span>
    </React.Fragment>
  ),
  errorMsg: '',
  handleClick: () => null,
  isPremiumProduct: false,
  isSelected: true,
  name: 'select 6',
  offerType: 'perc',
  perMonthAmtProps: {
    amount: 51500,
    currency: 'INR',
  },
  placeCart: () => null,
  price: 51500,
  productCode: 'PP_SP6',
  revampPage: false,
  saleprice: 51500,
  tagName: '',
  topAmountProps: {
    amount: 8584,
    currency: 'INR',
  },
};
const allProductsProps = {
  paymentTabsProps: {
    isPremiumProduct: true,
    productDetails: {
      currency: 'INR',
      currentTime: '1537424553',
      defaultProduct: 'SSP_GPlus',
      offerType: 'perc',
      products: {
        personalisedProducts: [
          {
            allow_discount: false,
            benefits: [
              {
                applicable: false,
                description: 'Dedicated Relationship Advisor',
                icon: 'dedicated',
                new_badge: false,
                tooltip: '',
              },
            ],
            best_value: false,
            default_selected: false,
            discount_code: '',
            discount_text: '',
            duration: 3,
            error_msg: '',
            name: 'select 3',
            price: 29500,
            pricepermonth: 9834,
            product_code: 'PP_SP1',
            product_subtext: '3 months',
            saleprice: 29500,
            topseller: false,
            your_plan: false,
          },
          {
            allow_discount: false,
            benefits: [
              {
                applicable: false,
                description: 'Dedicated Relationship Advisor',
                icon: 'dedicated',
                new_badge: false,
                tooltip: '',
              },
            ],
            best_value: false,
            default_selected: false,
            discount_code: '',
            discount_text: '',
            duration: 6,
            error_msg: '',
            name: 'select 6',
            price: 51500,
            pricepermonth: 8584,
            product_code: 'PP_SP6',
            product_subtext: '6 months',
            saleprice: 51500,
            topseller: false,
            your_plan: false,
          },
        ],
        premiumProducts: [
          {
            allow_discount: true,
            benefits: [],
            best_value: false,
            default_selected: false,
            discount_code: 'octbd60h2',
            discount_text: '50% off',
            duration: 3,
            error_msg: '',
            name: 'Gold',
            price: 4450,
            pricepermonth: 742,
            product_code: 'SSP_G3',
            product_subtext: '3 months',
            saleprice: 2225,
            topseller: false,
            your_plan: false,
          },
          {
            allow_discount: true,
            benefits: [],
            best_value: false,
            default_selected: true,
            discount_code: 'octbd60h2',
            discount_text: '55% off',
            duration: 3,
            error_msg: '',
            name: 'Gold Plus',
            price: 5550,
            pricepermonth: 833,
            product_code: 'SSP_GPlus',
            product_subtext: '3 months',
            saleprice: 2497,
            topseller: false,
            your_plan: false,
          },
          {
            allow_discount: true,
            benefits: [],
            best_value: false,
            default_selected: false,
            discount_code: 'octbd60h2',
            discount_text: '55% off',
            duration: 6,
            error_msg: '',
            name: 'Diamond',
            price: 6550,
            pricepermonth: 491,
            product_code: 'SSP_D6',
            product_subtext: '6 months',
            saleprice: 2947,
            topseller: false,
            your_plan: false,
          },
          {
            allow_discount: true,
            benefits: [],
            best_value: false,
            default_selected: false,
            discount_code: 'octbd60h2',
            discount_text: '60% off',
            duration: 6,
            error_msg: '',
            name: 'Diamond Plus',
            price: 8650,
            pricepermonth: 577,
            product_code: 'SSP_DPlus',
            product_subtext: '6 months',
            saleprice: 3460,
            topseller: true,
            your_plan: false,
          },
          {
            allow_discount: true,
            benefits: [],
            best_value: true,
            default_selected: false,
            discount_code: 'octbd60h2',
            discount_text: '60% off',
            duration: 12,
            error_msg: '',
            name: 'Platinum Plus',
            price: 14650,
            pricepermonth: 488,
            product_code: 'SSP_PPlus',
            product_subtext: '12 months',
            saleprice: 5860,
            topseller: false,
            your_plan: false,
          },
        ],
      },
      showSkipLink: false,
      showVip: true,
      skipProfileId: '',
      wwwBaseUrl: 'https://www.shaadi.com',
    },
    addToCart: () => null,
    cartResult: {
      btnloading: false,
      cartErrorMsg: '',
    },
    isRevampPage: true,
    isPersonalisedFeatures: false,
  },
  viewplan: true,
};
const allPersonalisedProductsProps = {
  paymentTabsProps: {
    isPremiumProduct: false,
    productDetails: {
      currency: 'INR',
      currentTime: '1537424553',
      defaultProduct: 'SSP_GPlus',
      offerType: 'perc',
      products: {
        personalisedProducts: [
          {
            allow_discount: false,
            benefits: [
              {
                applicable: false,
                description: 'Dedicated Relationship Advisor',
                icon: 'dedicated',
                new_badge: false,
                tooltip: '',
              },
            ],
            best_value: false,
            default_selected: false,
            discount_code: '',
            discount_text: '',
            duration: 3,
            error_msg: '',
            name: 'select 3',
            price: 29500,
            pricepermonth: 9834,
            product_code: 'PP_SP1',
            product_subtext: '3 months',
            saleprice: 29500,
            topseller: false,
            your_plan: false,
          },
          {
            allow_discount: false,
            benefits: [
              {
                applicable: false,
                description: 'Dedicated Relationship Advisor',
                icon: 'dedicated',
                new_badge: false,
                tooltip: '',
              },
            ],
            best_value: false,
            default_selected: false,
            discount_code: '',
            discount_text: '',
            duration: 6,
            error_msg: '',
            name: 'select 6',
            price: 51500,
            pricepermonth: 8584,
            product_code: 'PP_SP6',
            product_subtext: '6 months',
            saleprice: 51500,
            topseller: false,
            your_plan: false,
          },
        ],
        premiumProducts: [],
      },
      showSkipLink: false,
      showVip: true,
      skipProfileId: '',
      wwwBaseUrl: 'https://www.shaadi.com',
    },
    addToCart: () => null,
    cartResult: {
      btnloading: false,
      cartErrorMsg: '',
    },
    isRevampPage: true,
    isPersonalisedFeatures: false,
  },
  viewplan: true,
  viewPlansHandle: () => null,
};

const factory = {
  goldProps,
  goldPlusProps,
  diamondProps,
  diamondPlusProps,
  platinumPlusProps,
  yourPlanProps,
  selectThreeMonthProps,
  selectSixMonthProps,
  allProductsProps,
  allPersonalisedProductsProps,
};
export default factory;
