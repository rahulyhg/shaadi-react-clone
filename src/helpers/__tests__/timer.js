import React from 'react';
import renderer from 'react-test-renderer';
import Timer from '../timer';
import factory from './utils/factory';

describe('Timer : ', () => {
  const props = { ...factory.timerProps };
  describe('Timer Render: ', () => {
    const timerProps = {
      ...props,
      time: 2,
      loader: <div>loader</div>,
      response: <div>response!</div>,
    };

    it('Timer Redirect : Response ', done => {
      const tree = renderer.create(<Timer {...timerProps} />);
      expect(tree.toJSON()).toMatchSnapshot();
      setTimeout(() => {
        expect(tree.toJSON()).toMatchSnapshot();
        done();
      }, 2500);
    });
  });
});
