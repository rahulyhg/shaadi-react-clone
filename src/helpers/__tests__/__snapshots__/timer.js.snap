// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer :  Timer Render:  Timer Redirect : Response  1`] = `
<div>
  <div>
    loader
  </div>
</div>
`;

exports[`Timer :  Timer Render:  Timer Redirect : Response  2`] = `
<div>
  <div>
    response!
  </div>
</div>
`;
